packages/core/src/agent/MessageReceiver.ts

```diff
@@ -168,7 +168,7 @@ export class MessageReceiver {
     let message: AgentMessage
     try {
       message = await this.transformMessage(plaintextMessage)
-      await this.validateMessage(message)
+      this.validateMessage(message)
```

Comment:
`transformMessage` will also validate, so we're validating twice now. I think we can remove the `this.validateMessage` method completely

Reply:
Also this method uses `AgentMessage` for validation which looses the message specific properties, the validation done in the transform message method will cover this correctly